src/expressions/operator-expr.md

```diff
@@ -478,15 +478,48 @@ unsafe {
 assert_eq!(values[1], 3);
 ```
 
-#### Slice DST pointer to pointer cast
+#### Sized to sized pointer casts
 
-For slice types like `[T]` and `[U]`, the raw pointer types `*const [T]`, `*mut [T]`,
-`*const [U]`, and `*mut [U]` encode the number of elements in this slice. Casts between
-these raw pointer types preserve the number of elements. Note that, as a consequence,
-such casts do *not* necessarily preserve the size of the pointer's referent (e.g.,
-casting `*const [u16]` to `*const [u8]` will result in a raw pointer which refers to an
-object of half the size of the original). The same holds for `str` and any compound type
-whose unsized tail is a slice type, such as struct `Foo(i32, [u8])` or `(u64, Foo)`.
+A sized pointer to pointer cast `*const T as *const U` or `*mut T as *mut U`
+where `T: Sized` and `U: Sized` produces a value identical to the original
+pointer.
+
+#### Unsized to unsized pointer casts
+
+Any unsized to unsized pointer cast `*const T as *const U` or `*mut T as *mut U`
+produces a value identical to the original pointer if `T` and `U` are compatible
+unsized types.
+
+For slice types like `[T]` and `[U]`, the raw pointer types `*const [T]`, `*mut
+[T]`, `*const [U]`, and `*mut [U]` encode the number of elements in this slice.
+Casts between these raw pointer types preserve the number of elements. Note
+that, as a consequence, such casts do *not* necessarily preserve the size of the
+pointer's referent (e.g., casting `*const [u16]` to `*const [u8]` will result in
+a raw pointer which refers to an object of half the size of the original). The
+same holds for `str` and any compound type whose unsized tail is a slice type,
+such as struct `Foo(i32, [u8])` or `(u64, Foo)`.
+
+#### Unsized to sized pointer casts
+
+Any wide to narrow pointer cast discards all metadata that completes the wide
```

Comment:
The word "metadata" isn't defined anywhere. Can you add a description here that defines what it means?

Reply:
I don't know if I'm just promoting kicking the can down the road, but I opted not to define metadata since:
* It's not necessary to define the semantics of this cast, the only distinction is that unsized pointers have metadata and this cast removes it.
* Another part of the reference "behaviors considered undefined" uses metadata without defining it.

That being said, if I were to define it, where do you think I should do it and to what extent?

> The representation of pointers and references to DSTs is made up of two components; A base pointer and metadata. The metadata is the necessary extra data that has to be attached to pointers and references of DSTs which completes their representation. This is stored adjacent to the base pointer causing them to be wider than sized pointers earning them the name "wide" or "fat" pointers. The base pointer can be extracted through an unsized to sized pointer cast.

I'm a bit concerned that I'll end up having to define everything metadata adjacent 😓, so at least I'd want to avoid that as much as possible.